syncapi/storage/postgres/output_room_events_topology_table.go

```diff
@@ -115,10 +116,10 @@ func NewPostgresTopologyTable(db *sql.DB) (tables.Topology, error) {
 // on the event's depth.
 func (s *outputRoomEventsTopologyStatements) InsertEventInTopology(
 	ctx context.Context, txn *sql.Tx, event *gomatrixserverlib.HeaderedEvent, pos types.StreamPosition,
-) (err error) {
-	_, err = s.insertEventInTopologyStmt.ExecContext(
+) (topoPos types.StreamPosition, err error) {
```

Comment:
Ew, this means we rely on depth even more?

Reply:
Well, only in so far as that we're replicating it into the memberships table so that we can work out the topological bounds for `/messages`. If we stopped using depth for the topological position, we'd stop using it here too.